### DISCUSSION TITLE
How to use ESP32 pin as clock output?

### DISCUSSION THREAD

**Discussion by amirotin** [micropython/micropython#14036] [Core Development] [answered]
Hi. I want to use ESP32-S3 with SX1509 to control multiple fans using PWM. Unfortunately, it has a frequency of 2MHz, which can be reduced by using a divider using this formula - 2MHz / (2^(divider - 1). Thus, i can't get 25kHz used by fans.
SX1509 has an oscillatorPin, that can be used as external clock pin. Unfortunately i'm not very good at it yet and can't figure out exactly how to use it. Should i use simple [machine.PWM](https://docs.micropython.org/en/latest/library/machine.PWM.html#machine-pwm) or [esp32.RMT](https://docs.micropython.org/en/latest/library/esp32.html#esp32-rmt)?

**Comment by robert-hh** [comment, marked as answer]
You can easily use PWM for that purpose. I just verified that at an ESP32-S3 with 8 Pins at exactly 25kHz and different duty cycles.

![grafik](https://github.com/micropython/micropython/assets/12476868/1e2c2241-6560-408e-bb45-0e9393709d43)
